core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -392,15 +396,22 @@ public IndicesQueriesRegistry buildQueryParserRegistry() {
 
         for (QueryRegistration<?> query : queries) {
             QueryParser<? extends QueryBuilder<?>> parser = query.parser;
-            for (String name: query.names) {
+            for (String name: query.parseField.getAllNamesIncludedDeprecated()) {
```

Comment:
Sounds good. I was also wondering if we need the QueryRegistration object since I think we should be able to directly register the query with the named writeable registry and the parser registry in the register method